<a href="https://colab.research.google.com/github/Sobrero22/Python/blob/master/FelipeSobrero_Aula3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Exercício 1)Crie uma funcão que tome um argumento e imprima o valor e o tipo dele.

In [ ]:
def verso(poema) :
    print(poema)
    print('tipo de argumento',type(poema))

verso('voar eu não vou nunca mais')

voar eu não vou nunca mais
tipo de argumento <class 'str'>


Exercício 2)Crie uma função que calcule e imprima velocidade media de um objeto a partir de uma posição inicial, a final e o tempo transcorrido para um objeto em MRU. Também crie uma funcão que calcule e imprima a velocidade de um objeto a partir da aceleração constante e o tempo (MRUA) (p.ex. queda libre)

In [ ]:
x_i = float(input('Digite a posição inicial: '))
x_f = float(input('Digite a posição final: '))
a = float(input('Digite a aceleração: '))
t = float(input('Digite o tempo transcorrido: '))

if a == 0 :
  def f1(x_i,x_f,t) : 
    vm = (x_f-x_i)/t
    return vm

  velocidade = f1(x_i,x_f,t) 
  print('A velocidade média é', velocidade, 'm/s')

else :
  def f2(a,t) :
    v = a*t
    return v

  velocidade = f2(a,t) 
  print('A velocidade é', velocidade, 'm/s')

Digite a posição inicial: 100
Digite a posição final: 0
Digite a aceleração: 9.81
Digite o tempo transcorrido: 10
A velocidade é 98.10000000000001 m/s


Exercício 3)Crie uma funcão para calcular o ángulo zenital do sol (da semana passada) tomando como argumento as medidas da altura e o comprimento da sombra.

In [ ]:
h = float(input('Digite a altura: '))
s = float(input('Digite o comprimento da sombra: '))

import math

def f(h,s) :
  theta = round(math.degrees(math.atan(s/h)),2)
  return theta

theta = f(h,s)

print('O angulo zenital é', theta, '°')

Digite a altura: 10
Digite o comprimento da sombra: 18
O angulo zenital é 60.95 °


Exercício 4)Crie uma função que faça a conversão de uma medida inicialmente em milhas para m, e outra para o inverso; uma de horas para segundos, e o inverso. Utilize estas funções para resolver novamente o primeiro exercício da semana passada (da corrida). Se uma pessoa demora 30 minutos em 4 milhas, qual velocidade media em km/h ? e o tempo medio por kilometro?

In [ ]:
answer = input('Sua medida está em metros ou em milhas? ')

if answer=='metros':
  metros = float(input('Digite sua medida: '))
  
  def converte_metros_em_milha(metros):
    milhas = metros/1609.34
    return milhas

  milhas = converte_metros_em_milha(metros)
  print('Sua medida em milhas é:', milhas)

if answer=='milhas':
  milhas = float(input('Digite sua medida:'))
  
  def converte_milhas_em_metros(milhas):
    metros = milhas*1609.34
    return metros

  metros = converte_milhas_em_metros(milhas)
  print('Sua medida em metros é:', metros)


Sua medida está em metros ou em milhas? milhas
Digite sua medida:4
Sua medida em metros é: 6437.36


In [ ]:
answer = input('Sua medida está em horas ou em segundos? ')

if answer=='horas':
  horas = float(input('Digite sua medida: '))
  
  def converte_horas_em_segundos(horas):
    segundos = horas*3600
    return segundos

  segundos = converte_horas_em_segundos(horas)
  print('Sua medida em segundos é:', segundos)

if answer=='segundos':
  segundos = float(input('Digite sua medida:'))
  
  def converte_segundos_em_horas(segundos):
    horas = segundos/3600
    return horas

  horas = converte_segundos_em_horas(segundos)
  print('Sua medida em metros é:', horas)


Sua medida está em horas ou em segundos? horas
Digite sua medida: 0.07767159208122584
Sua medida em segundos é: 279.61773149241304


In [ ]:
6437.36/(1000*0.5)

12.87472

In [ ]:
1/12.87472

0.07767159208122584

Sua velocidade será 12.87 km/h.E o tempo médio será 279.62 s/km.

Exercício 5)Crie funções para calcular os outros exemplos das aulas anteriores: IMC, volume de uma esfera, distancia entre pontos de máximos de difração. Decida quais serão os argumentos e o valor retornado.

In [ ]:
massa = float(input('Qual a sua massa (em kg)?'))
altura = float(input('Qual a sua altura (em metros)?'))

def f(massa,altura):
  IMC = round(massa/(altura**2),2)
  return IMC

IMC=f(massa,altura)

answer=input('Você é homem ou mulher?')

if IMC <18 :
  print('Seu IMC é',IMC, '. Você está abaixo do peso.')

elif answer=='mulher' and IMC >=18 and IMC <=24:
  print('Seu IMC é',IMC, '. Você tem um IMC considerado normal.')

elif answer=='homem' and IMC >=18 and IMC <=25:
  print('Seu IMC é',IMC, '. Você tem um IMC considerado normal.')

else:
  print('Seu IMC é',IMC, '. Você está com sobrepeso.')

Qual a sua massa (em kg)?80
Qual a sua altura (em metros)?1.8
Você é homem ou mulher?mulher
Seu IMC é 24.69 . Você está com sobrepeso.


In [ ]:
raio=float(input('Qual o raio da esfera (em metros)? '))

import math

def f(raio):
  volume = round((4/3)*math.pi*(raio**3),2)
  return volume

volume = f(raio)
print('O volume da esfera é',volume,'m³')

Qual o raio da esfera (em metros)? 5
O volume da esfera é 523.6 m³


In [ ]:
lamb=float(input('Qual o comprimento de onda do laser (dado em nanometros)? '))
D=float(input('Qual a distancia do anteparo (dado em metros)? '))
d=float(input('Qual a o espaçamento entre as fendas (dado em milimetros)? '))

def f3(lamb,D,d):
  max_de_difracao=lamb*D*1000/((10**9)*d)
  return max_de_difracao

max_de_difracao=f3(lamb,D,d)
print('A distancia entre dois máximos consecutivos será',max_de_difracao,'m.')

Qual o comprimento de onda do laser (dado em nanometros)? 530
Qual a distancia do anteparo (dado em metros)? 2
Qual a o espaçamento entre as fendas (dado em milimetros)? 10
A distancia entre dois máximos consecutivos será 0.000106 m.
